src/main/java/uk/gov/hmcts/reform/document/DocumentUploadClientApi.java

```diff
@@ -51,10 +52,11 @@ public UploadResponse upload(
         @RequestHeader(HttpHeaders.AUTHORIZATION) String authorisation,
         @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuth,
         @RequestHeader(USER_ID) String userId,
+        @RequestHeader(ROLES) String roles,
```

Comment:
might want to consider a couple of overloaded methods here so that users don't need to pass null around

uploadWithRoles
uploadWithUserId

also are these request header annotations actually needed?

and one more also, a String of roles is probably unlikely, either a var args method or a list would probably be more appropriate

Reply:
As we are passing roles in request `parameter` to doc store, shouldn't we use RequestParam here as well rather than RequestHeader? 
Also shouldn't this be a Collection of roles as per below?
https://github.com/hmcts/document-management-store-app/blob/master/application/src/main/java/uk/gov/hmcts/dm/commandobject/UploadDocumentsCommand.java#L32

Reply:
done